packages/xo-server/src/xo-mixins/authorization.mjs

```diff
@@ -0,0 +1,78 @@
+import { createLogger } from '@xen-orchestra/log'
+import get from 'lodash/get.js'
+import { featureUnauthorized } from 'xo-common/api-errors.js'
+import assert from 'assert'
+
+const { warn } = createLogger('xo:server:authorization')
+
+const FREE = 1
+const STARTER = 2
+const ENTREPRISE = 3
+const PREMIUM = 4
+const OPEN = 5
+
+export const TRIAL_LEVELS = {
+  free: FREE,
+  starter: STARTER,
+  entreprise: ENTREPRISE,
+  premium: PREMIUM,
+  open: OPEN,
+}
+
+const AUTHORIZATIONS = {
+  BACKUP: {
+    DELTA: STARTER,
+    DELTA_REPLICATION: ENTREPRISE,
+    FULL: STARTER,
+    METADATA: ENTREPRISE,
+    WITH_RAM: ENTREPRISE,
+  },
+  DOCKER: STARTER, // @todo  _doDockerAction in xen-orchestra/packages/xo-server/src/xapi/index.mjs
+  EXPORT: {
+    XVA: STARTER, // @todo handleExport in xen-orchestra/packages/xo-server/src/api/vm.mjs
+  },
+}
+
+export default class Authorization {
+  #app
+  constructor(app) {
+    this.#app = app
+  }
+
+  #getMinPlan(featureCode) {
+    const minPlan = get(AUTHORIZATIONS, featureCode)
+    assert.notEqual(minPlan, undefined, `${featureCode} is not defined in the AUTHORIZATIONS object`)
+    return minPlan
+  }
+
+  async #getCurrentPlan() {
+    if (this.#app.getXoaPlan === undefined) {
+      // source user => everything is open
+      return OPEN
```

Comment:
What does `OPEN` mean? Is it open source? Or open as available?

It's a bit confusing :smile: